### PR TITLE
Use doctrine registry to cope with multiple managers

### PR DESCRIPTION
**Pull request by mduplouy**
Hi there,

This fix a MappingException (class not found in chain configured namespaces) when using several entity managers

### CODE REVIEW

File: src/Repository/DefaultSluggableRepository.php [hunk 1]
```diff
@@ -4,26 +4,28 @@
...
+        $this->registry = $registry;
     }
 
     public function isSlugUniqueFor(SluggableInterface $sluggable, string $uniqueSlug): bool
     {
         $entityClass = get_class($sluggable);
 
-        $queryBuilder = $this->entityManager->createQueryBuilder()
+        $queryBuilder = $this->registry
+            ->getManagerForClass($entityClass)
```
Comment (TomasVotruba) [inline src/Repository/DefaultSluggableRepository.php]: Better decopule this logic to standalone line, as it is easy to oversee.

```php
$sluggableRepository = $this->registry->getManagerForClass($entityClass);
```

Also, this change should be done probably or all subscribers, right?

Comment (mduplouy) [reply src/Repository/DefaultSluggableRepository.php]: Just trying to get all test jobs ok before making any other modification :)

any idea on how to solve this one ? https://travis-ci.org/KnpLabs/DoctrineBehaviors/jobs/638486665

Do i really need to add a condition on the manager instance ?

Comment (TomasVotruba) [reply src/Repository/DefaultSluggableRepository.php]: @var annotation above the variable should solve that

**Comment by mduplouy** [comment]
I finally made it :)

@TomasVotruba i also modified other subscribers, let me know if anything has to be modified

File: src/Repository/DefaultSluggableRepository.php [hunk 2]
```diff
@@ -5,25 +5,29 @@
...
+        $this->managerRegistry = $managerRegistry;
     }
 
     public function isSlugUniqueFor(SluggableInterface $sluggable, string $uniqueSlug): bool
     {
         $entityClass = get_class($sluggable);
 
-        $queryBuilder = $this->entityManager->createQueryBuilder()
+        /** @var EntityManagerInterface $manager */
+        $manager = $this->managerRegistry->getManagerForClass($entityClass);
```
Comment (TomasVotruba) [inline src/Repository/DefaultSluggableRepository.php]:
```suggestion
        $entityManager = $this->managerRegistry->getManagerForClass($entityClass);
```

This would be more descriptive name :+1:

File: config/services.yaml [hunk 3]
```diff
@@ -18,3 +18,5 @@ services:
         exclude:
             - "../src/Bundle/*"
             - "../src/Exception/*"
+
+    Doctrine\Persistence\ManagerRegistry: '@doctrine'
```
Comment (TomasVotruba) [inline config/services.yaml]: Why is this needed? It might break the app that is using this bundle

Comment (mduplouy) [reply config/services.yaml]: without this the autowire does not work well with some versions :/ 

https://travis-ci.org/KnpLabs/DoctrineBehaviors/jobs/638475665?utm_medium=notification&utm_source=github_status

Comment (TomasVotruba) [reply config/services.yaml]: I think it should be compatible with both. 
Moving alias to a services that's there in some conditions might break stuff.

Comment (TomasVotruba) [reply config/services.yaml]: Standalone test case for ManagerRegistry with own would definitelly help

Comment (mduplouy) [reply config/services.yaml]: Sorry but I don't understand what you mean
 imo autowiring should not be used and doctrine service explicitely injected

Comment (mduplouy) [reply config/services.yaml]: ok
apologize i though doctrine bundle wasn't a dev dependency
i found the weird alias stuff -- with if (false) condition ?? -- and type hinted on it, everything seems ok

**Comment by mduplouy** [comment]
@TomasVotruba is this ok for you ?

**Comment by TomasVotruba** [comment]
Code looks good. 
It needs test case with 2 entity managers to prove it works correctly

**Comment by TomasVotruba** [comment]
Closing as stale
